http-clients/aws-crt-client/src/test/java/software/amazon/awssdk/http/crt/ProxyConfigurationTest.java

```diff
@@ -27,12 +27,45 @@
  */
 public class ProxyConfigurationTest {
     private static final Random RNG = new Random();
+    private static final String TEST_HOST = "foo.com";
+    private static final int TEST_PORT = 7777;
+    private static final String TEST_USER = "testuser";
+    private static final String TEST_PASSWORD = "123";
 
     @Test
     public void build_setsAllProperties() {
         verifyAllPropertiesSet(allPropertiesSetConfig());
     }
 
+    @Test
+    public void build_systemPropertyEnabled() {
+        setSystemProperty();
```

Comment:
Can we also clear all system properties before and after a text execution? E.g.,

https://github.com/aws/aws-sdk-java-v2/blob/9f8e42b08a34cc32d7ba7ac92b340be8153de50c/http-clients/apache-client/src/test/java/software/amazon/awssdk/http/apache/ProxyConfigurationTest.java#L29-L37

This will help make sure our test results aren't dependent upon the test execution order (especially future tests).

Reply:
I also see we have some unit test failures in ` NettyClientTlsAuthTest`, ` AwaitCloseChannelPoolMapTest`, and ` ProxyWireMockTest`. These classes are all using `ProxyConfiguration` and may be failing due to the leftover system properties.